core/environment/environment.go

```diff
@@ -216,17 +229,43 @@ func newEnvironment(userVars map[string]string, newId uid.ID) (env *Environment,
 				} else if e.Event == "STOP_ACTIVITY" {
 					endTime, ok := env.workflow.GetUserVars().Get("run_end_time_ms")
 					if ok && endTime == "" {
-						runEndTime := strconv.FormatInt(time.Now().UnixMilli(), 10)
-						env.workflow.SetRuntimeVar("run_end_time_ms", runEndTime)
+						runEndTime := time.Now()
+						runEndTimeS := strconv.FormatInt(runEndTime.UnixMilli(), 10)
+						env.workflow.SetRuntimeVar("run_end_time_ms", runEndTimeS)
+						env.workflow.SetRuntimeVar("run_start_time_ms", "") // we delete previous SOSOR
+
+						the.EventWriterWithTopic(topic.Run).WriteEventWithTimestamp(&pb.Ev_RunEvent{
+							EnvironmentId:    envId.String(),
+							RunNumber:        env.GetCurrentRunNumber(),
+							State:            env.Sm.Current(),
+							Error:            "",
+							Transition:       e.Event,
+							TransitionStatus: pb.OpStatus_STARTED,
+							Vars:             nil,
+						}, runEndTime)
+
 					} else {
 						log.WithField("partition", envId.String()).
 							Debug("O2 End time already set before before_STOP_ACTIVITY")
 					}
 				} else if e.Event == "GO_ERROR" {
 					endTime, ok := env.workflow.GetUserVars().Get("run_end_time_ms")
 					if ok && endTime == "" {
-						runEndTime := strconv.FormatInt(time.Now().UnixMilli(), 10)
-						env.workflow.SetRuntimeVar("run_end_time_ms", runEndTime)
+						runEndTime := time.Now()
+						runEndTimeS := strconv.FormatInt(runEndTime.UnixMilli(), 10)
+						env.workflow.SetRuntimeVar("run_end_time_ms", runEndTimeS)
+						env.workflow.SetRuntimeVar("run_start_time_ms", "") // we delete previous SOSOR
```

Comment:
Why delete it here? It seems reasonable to me that tasks transitioning from RUNNING to CONFIGURED (due something else going to ERROR) might want to get all the known timestamps of the finishing run.
Perhaps you intended to delete EOEOR here?